### DISCUSSION TITLE
Any way to hide the left sidebar?

### DISCUSSION THREAD

**Discussion by onurusluca** [withastro/starlight#1108] [Q&A] [answered]
### What version of `starlight` are you using?

0.11.2

### What is your idea?
Forgive my ignorance if there is already a way to do this.
Would be greate to have an option to hide the left sidebar. Or much better, a way to toggle it!

```
---
title: "Guided Setup"
description: "Guided Setup for Custom Element"
leftSidebar: false // or `toggle`
---
```

### Why is this feature necessary?

On some pages, some might want to have a clean page only.

### Participation

- [X] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment, marked as answer]
The [`template: splash` option](https://starlight.astro.build/guides/customization/#page-layout) hides the left sidebar:

```md
---
title: Page without sidebars
template: splash
---
```

It does also hide the right sidebar/table of contents though. Were you hoping to keep that?

**Reply by onurusluca** [reply]
Oh thanks! That actually works for me. Couldn't find it in the docs. Just in case someone else wondered about this: [docs link](https://starlight.astro.build/reference/frontmatter/#template)
```
Set the layout template for this page. Pages use the 'doc' layout by default. Set to 'splash' to use a wider layout without any sidebars designed for landing pages.
```

**Reply by butcherof** [reply]
Is there a way to hide the left sidebar, but keep the right sidebar-TOC?

**Reply by Muzych** [reply]
Is there any way for users to control this?

**Reply by j0sh** [reply]
Would also like to know how to hide the left sidebar, but keep the table of contents on the right.

**Reply by KaiSpencer** [reply]
You can just hide the sidebar (left) with the following

```yml
---
sidebar:
    hidden: true
---
```
[Docs](https://starlight.astro.build/reference/frontmatter/#sidebar)

**Reply by j0sh** [reply]
@KaiSpencer Thanks - that option does hide the text of the left sidebar, but keeps the space for it, along with the border outline - see screenshot below. The overall effect is a little odd, it would be nicer to remove the sidebar completely
 
<img width="920" alt="image" src="https://github.com/withastro/starlight/assets/292510/49fa3933-2f25-4eca-95ae-9ecc7f59e15a">

**Reply by yangshaohongit** [reply]
> @KaiSpencer Thanks - that option does hide the text of the left sidebar, but keeps the space for it, along with the border outline - see screenshot below. The overall effect is a little odd, it would be nicer to remove the sidebar completely
> 
> <img alt="image" width="920" src="https://private-user-images.githubusercontent.com/292510/313767471-49fa3933-2f25-4eca-95ae-9ecc7f59e15a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM5OTYzMTksIm5iZiI6MTc0Mzk5NjAxOSwicGF0aCI6Ii8yOTI1MTAvMzEzNzY3NDcxLTQ5ZmEzOTMzLTJmMjUtNGVjYS05NWFlLTllY2M3ZjU5ZTE1YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQwN1QwMzIwMTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kMTQxZGM4ZWE4Y2FkMzc4MWY2YjkwNGIzYWMyOTc4OTc0ZDFjZmFhZjQ5M2FiNDk1OTU1ZDJjYmYyODVlNWQwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.PPctkxddw7GzW8-iRfP2hdhYENumVDEBryoEP454F3E">

😁Same question，anyone help? How to hide the border outline too?

**Reply by Judimax** [reply]
https://windmillcode.github.io/starlight-fullview-mode/ might be even a better solution than what yr looking for

**Reply by Taewan-P** [reply]
@Judimax Any way of just keep hiding the left sidebar only? Without toggles?

**Reply by Judimax** [reply]
Like complete collapse  or just remove the toggle schedule time on our calendar to discuss what you are trying to do https://windmillcode.com/contact @Taewan-P

**Reply by Taewan-P** [reply]
@Judimax Just complete collapse. I just want to remove it from the screen unless it's in mobile UI.

**Reply by Judimax** [reply]
We have made the changes for you enjoy!
https://windmillcode.github.io/starlight-fullview-mode/configuration/#collapse-the-sidebars-from-view
@Taewan-P

**Reply by Taewan-P** [reply]
@Judimax Wow... Thanks for the super fast update. It's like one of the fastest feedback I've ever experienced.

**Reply by Judimax** [reply]
@Taewan-P its personal for me getting slow updates to a library at Windmillcode that's what makes us different from the competition

**Comment by abraark40** [comment]
Hey just a test comment

**Comment by mb21** [comment]
In addition to the [template: splash](https://starlight.astro.build/guides/customization/#page-layout) option delucis mentioned:

```md
---
title: Page without sidebars
template: splash
---
```

I added a [custom stylesheet](https://starlight.astro.build/guides/css-and-tailwind/#custom-css-styles) with:

```css
html:not([data-has-toc]) {
  --sl-content-width: 45rem !important;
}
```

**Comment by Judimax** [comment]
https://windmillcode.github.io/starlight-fullview-mode/

**Comment by HiDeoo** [comment]
Considering many suggestions have been listed to handle this use case, either with a different template, CSS-solutions, or even plugins, depending on the desired outcome, I'll close this discussion for now.

**Comment by shubham-padia** [comment]
Would starlight be open to a PR that makes having a collapsible sidebar part of the codebase, it can be similar to https://windmillcode.github.io/starlight-fullview-mode/ mentioned above, but with design that looks more coherent to the codebase.

We could develop our own plugin, but plugins do become harder and harder to maintain with more and more upstream changes and this feels like a feature that would be beneficial for all users (not a niche use case which warrants this to be limited to a plugin).
